beluga/include/beluga/algorithm/estimation.hpp

```diff
@@ -169,9 +171,50 @@ std::pair<Sophus::SE2<Scalar>, Sophus::Matrix3<Scalar>> estimate(Poses&& poses,
   return std::pair{estimated_pose, covariance_matrix};
 }
 
-/// Computes mean and covariance Returns a pair consisting of the estimated mean pose and its covariance.
+/// Computes mean and standard deviation of a range of weighted scalars.
 /**
- * Given a range of 2D poses, computes the estimated pose by averaging the translation
+ * Given a range of scalars, computes the scalar mean and standard deviation.
+ *
+ * \tparam Scalars A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `std::vector<Scalar>`.
+ * \tparam Weights A [sized range](https://en.cppreference.com/w/cpp/ranges/sized_range) type whose
+ *  value type is `Scalar`.
+ * \tparam Scalar The scalar value type of the given range of Scalars.
+ * \param scalars Range of scalars.
+ * \param weights Range of weights.
+ * \return The estimated mean and its standard deviation.
+ */
+template <
+    class Scalars,
+    class Weights,
+    class Scalar = ranges::range_value_t<Scalars>,
+    typename = std::enable_if_t<std::is_arithmetic_v<Scalar>>>
+std::pair<Scalar, Scalar> estimate(Scalars&& scalars, Weights&& weights) {
```

Comment:
Sorry for what seems to be a nitpick, but could you please add a unit tests for this code as it will be part of the algorithms library?

Reply:
Yes, sure!

Reply:
@nahueespinosa The task is done, and you can check the last commit.
I've verified it by running `colcon test --packages-select beluga`.

Reply:
Neat!